cmd/deployment/resource/stop.go

```diff
@@ -0,0 +1,112 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package cmddeploymentresource
+
+import (
+	"fmt"
+
+	"github.com/elastic/cloud-sdk-go/pkg/util/ec"
+	"github.com/spf13/cobra"
+
+	cmdutil "github.com/elastic/ecctl/cmd/util"
+	"github.com/elastic/ecctl/pkg/deployment"
+	"github.com/elastic/ecctl/pkg/deployment/depresource"
+	"github.com/elastic/ecctl/pkg/ecctl"
+)
+
+var stopCmd = &cobra.Command{
+	Use:     "stop <deployment id> --type <type> [--all|--i <instance-id>,<instance-id>]",
+	Short:   "Stops a deployment resource",
+	PreRunE: cmdutil.MinimumNArgsAndUUID(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		resType, _ := cmd.Flags().GetString("type")
+		refID, _ := cmd.Flags().GetString("ref-id")
+		instanceID, _ := cmd.Flags().GetStringSlice("instance-id")
+		ignoreMissing, _ := cmd.Flags().GetBool("ignore-missing")
+
+		if refID == "" {
+			var err error
+			refID, err = getRefID(resType, args[0])
+			if err != nil {
+				return err
+			}
+		}
+
+		if err := cmdutil.IncompatibleFlags(cmd, "all", "instance-id"); err != nil {
+			fmt.Fprintln(cmd.OutOrStderr(), err)
+		}
+
+		if all, _ := cmd.Flags().GetBool("all"); all {
+			_, err := depresource.Stop(depresource.StopParams{
+				API:          ecctl.Get().API,
+				DeploymentID: args[0],
+				Type:         resType,
+				RefID:        refID,
+			})
+			if err != nil {
+				return err
+			}
+
+			return nil
+		}
+
+		_, err := depresource.StopInstances(depresource.StopInstancesParams{
+			StopParams: depresource.StopParams{
+				API:          ecctl.Get().API,
+				DeploymentID: args[0],
+				Type:         resType,
+				RefID:        refID,
+			},
+			InstanceIDs:   instanceID,
+			IgnoreMissing: ec.Bool(ignoreMissing),
+		})
+		if err != nil {
+			return err
+		}
+
+		return nil
+
+	},
+}
+
+func getRefID(resType, depID string) (string, error) {
```

Comment:
I don't think this belongs to the cmd layer. Should be moved to the `pkg` and more specifically under the deployment package. can be used by other operations and should be also unit tested